src/streamlink/plugins/dogus.py

```diff
@@ -1,57 +1,43 @@
-        kralmuzik\.com\.tr/tv/|
-        ntv\.com\.tr/canli-yayin/ntv|
-        startv\.com\.tr/canli-yayin
-    )/?
-""", re.VERBOSE))
+@pluginmatcher(re.compile(r"https?://(?:www\.)?eurostartv\.com\.tr/canli-izle"))
+@pluginmatcher(re.compile(r"https?://(?:www\.)?kralmuzik\.com\.tr/tv/.+"))
+@pluginmatcher(re.compile(r"https?://(?:www\.)?ntv\.com\.tr/canli-yayin/ntv"))
+@pluginmatcher(re.compile(r"https?://(?:www\.)?startv\.com\.tr/canli-yayin"))
 class Dogus(Plugin):
-    mobile_url_re = re.compile(r"""(?P<q>["'])(?P<url>(https?:)?//[^'"]*?/live/hls/[^'"]*?\?token=)
-                                   (?P<token>[^'"]*?)(?P=q)""", re.VERBOSE)
-    token_re = re.compile(r"""token=(?P<q>["'])(?P<token>[^'"]*?)(?P=q)""")
-    kral_token_url = "https://service.kralmuzik.com.tr/version/gettoken"
+    _re_live_hls = re.compile(r"'(https?://[^']+/live/hls/[^']+)'")
+    _re_yt_script = re.compile(r"youtube\.init\('([\w-]{11})'")
 
     def _get_streams(self):
-        res = self.session.http.get(self.url)
-
-        # Look for Youtube embedded video first
-        for iframe in itertags(res.text, "iframe"):
-            if urlparse(iframe.attributes.get("src")).netloc.endswith("youtube.com"):
-                log.debug("Handing off to YouTube plugin")
-                return self.session.streams(iframe.attributes.get("src"))
-
-        # Next check for HLS URL with token
-        mobile_url_m = self.mobile_url_re.search(res.text)
-        mobile_url = mobile_url_m and update_scheme("https://", mobile_url_m.group("url"), force=False)
-        if mobile_url:
-            log.debug("Found mobile stream: {0}".format(mobile_url_m.group(0)))
-
-            token = mobile_url_m and mobile_url_m.group("token")
-            if not token and "kralmuzik" in self.url:
-                log.debug("Getting Kral Muzik HLS stream token from API")
-                token = self.session.http.get(self.kral_token_url).text
-            elif not token:
-                # if no token is in the url, try to find it else where in the page
-                log.debug("Searching for HLS stream token in URL")
-                token_m = self.token_re.search(res.text)
-                token = token_m and token_m.group("token")
-
-            return HLSStream.parse_variant_playlist(self.session, mobile_url + token,
-                                                    headers={"Referer": self.url})
+        root = self.session.http.get(self.url, schema=validate.Schema(validate.parse_html()))
+
+        # https://www.ntv.com.tr/canli-yayin/ntv?youtube=true
+        yt_iframe = root.xpath("string(.//iframe[contains(@src,'youtube.com')]/@src)")
+        # https://www.startv.com.tr/canli-yayin
+        dm_iframe = root.xpath("string(.//iframe[contains(@src,'dailymotion.com')]/@src)")
+        # https://www.kralmuzik.com.tr/tv/kral-tv
+        # https://www.kralmuzik.com.tr/tv/kral-pop-tv
+        yt_script = root.xpath("string(.//script[contains(text(), 'youtube.init')]/text())")
```

Comment:
It's probably better to add a `[1]` XPath predicate to each node, because otherwise it could return `@src`/`text()` values of multiple nodes. So for example
```
string(.//iframe[contains(@src,'youtube.com')][1]/@src)
```

Same in the XPath query down below.